core/src/subgraph/instance_manager.rs

```diff
@@ -508,7 +501,18 @@ where
                 block.clone(),
                 triggers,
             )
-            .map(move |block_state| (ctx, block_state, data_sources, runtime_hosts))
+            .and_then(move |block_state| {
+                match block_state.created_data_sources.len() > created_ds_count {
+                    false => Ok((ctx, block_state, data_sources, runtime_hosts)),
+                    true => Err(err_msg(
+                        "A dynamic data source, in the same block that it was created,
+                        attempted to create another dynamic data source.
+                        To let us know that you are affected by this bug, please comment in
+                        https://github.com/graphprotocol/graph-node/issues/1105",
```

Comment:
Nice, I like the approach of pushing users to a specific issue if they encounter the problem.